src/UI/Dependency/EnvironmentModifier.tsx

```diff
@@ -36,10 +36,14 @@ export function useEnvironmentModifierImpl(): EnvironmentModifier {
   }
 
   function useSetting(settingName: keyof EnvironmentSettings.DefinitionMap): boolean {
```

Comment:
Can you add some comments about the problems we are facing now? To keep them in the code.

Reply:
added mention in the docstrings for this function, created task to resolve warning and errors where I mentioned re-rendering and mentioned that issue in the GraphQL task  and linked this task to the docstrings